src/pages/_app.jsx

```diff
@@ -33,6 +86,8 @@ const WrappedApp = ({ Component, pageProps }) => {
     if (router.route.includes('experimentId')) {
       setExperimentId(router.query.experimentId);
     }
+
+    setupAmplify();
```

Comment:
move this somewhere else, it doesn't depend on the experimentId and leave comments on why it is here

Reply:
turns out the checks in this useEffect are useful for setupAmplify, so it might be ok to keep it here